### PR TITLE
GEN-1397 - feat(widget): added checkout page

### PR DESCRIPTION
**Pull request by guilhermespopolin**
## Describe your changes

- Added Widget Checkout page

## Justify why they are needed

Foundation for Widget Checkout page. Other aspects of it like localisation and rendering CMS content will be addressed on subsequent PRs.

### CODE REVIEW

**Comment by guilhermespopolin** [comment]
Current dependencies on/for this PR:
* main
  * **PR #3554** <a href="https://app.graphite.dev/github/pr/HedvigInsurance/racoon/3554?utm_source=stack-comment-icon" target="_blank">Graphite</a>  👈

This [stack of pull requests](https://stacking.dev/?utm_source=stack-comment) is managed by [Graphite](https://app.graphite.dev/github/pr/HedvigInsurance/racoon/3554?utm_source=stack-comment).

File: apps/store/src/pages/widget/[flow]/[shopSessionId]/checkout.tsx [hunk 1]
```diff
@@ -0,0 +1,86 @@
...
+      return fallbackRedirect
+    }
+
+    if (!customer.ssn) {
+      console.warn('Widget Checkout | No SSN in shop session', shopSession.id)
+      return fallbackRedirect
+    }
+
+    return {
+      props: {
```
Comment (robinandeer) [inline apps/store/src/pages/widget/[flow]/[shopSessionId]/checkout.tsx:68]: We should pre-populate Apollo cache here

File: apps/store/src/pages/widget/[flow]/[shopSessionId]/checkout.tsx [hunk 2]
```diff
@@ -0,0 +1,86 @@
...
+export const getServerSideProps: GetServerSideProps<Props, Params> = async (context) => {
+  if (!context.params) throw new Error('Missing params')
+  if (!isRoutingLocale(context.locale)) throw new Error(`Invalid locale: ${context.locale}`)
+
+  const fallbackRedirect = {
+    redirect: {
+      destination: PageLink.home({ locale: context.locale }).toString(),
+      permanent: false,
+    },
+  } as const
```
Comment (robinandeer) [inline apps/store/src/pages/widget/[flow]/[shopSessionId]/checkout.tsx:41]: I understand it's a work in progress but this seems wrong to me

We need to display a better error state to the user instead -- let's just throw error instead?

File: apps/store/src/features/widget/CheckoutPage.tsx [hunk 3]
```diff
@@ -0,0 +1,255 @@
...
+
+        <GridLayout.Root>
+          <GridLayout.Content width="1/3" align="center">
+            <Space y={3.5}>
+              <Headings>
+                <Heading as="h1" variant="standard.24" align="center">
+                  {/* TODO: translate this */}
+                  Ditt erbjudande
+                </Heading>
+                <Heading
```
Comment (robinandeer) [inline apps/store/src/features/widget/CheckoutPage.tsx:91]: This should probably just be text paragraph

**Review by robinandeer — APPROVED** [review]
Nice work! Let's discuss what/how we can reuse from regular checkout

Added via Giphy

File: apps/store/src/pages/widget/[flow]/[shopSessionId]/checkout.tsx (hunk 2 above)
Comment (guilhermespopolin) [reply apps/store/src/pages/widget/[flow]/[shopSessionId]/checkout.tsx:41]: Yes, ofc.

**Pull request merged**